src/spikeinterface/extractors/phykilosortextractors.py

```diff
@@ -172,7 +172,14 @@ def __init__(
                 if load_all_cluster_properties:
                     # pandas loads strings as objects
                     if cluster_info[prop_name].values.dtype.kind == "O":
-                        prop_dtype = type(cluster_info[prop_name].values[0])
+                        for value in cluster_info[prop_name].values:
+                            if isinstance(value, (np.floating, float)) and np.isnan(
+                                value
+                            ):  # Blank values are encoded as 'NaN'.
+                                continue
+
+                            prop_dtype = type(value)
+                            break
```

Comment:
what if they're all NaN?

Reply:
Then the dtype wouldn't be object, but float?
So you never reach this code (I think)

Reply:
I think I'm being dense. Why do you need both checks in the if? Why not just check if `isnan`? Or flip the order of the if so that if it is not nan we short-circuit? Small point but I'm just trying to figure out why we need both.

Reply:
Perfect thanks!